pubsub/query/query_test.go

```diff
@@ -0,0 +1,43 @@
+package query_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tmlibs/pubsub/query"
+)
+
+func TestMatches(t *testing.T) {
+	testCases := []struct {
+		s       string
+		tags    map[string]interface{}
+		err     bool
+		matches bool
+	}{
+		{"tm.events.type=NewBlock", map[string]interface{}{"tm.events.type": "NewBlock"}, false, true},
+
+		{"tx.gas > 7", map[string]interface{}{"tx.gas": 8}, false, true},
```

Comment:
okay, i guess this answers my questions above.  if there is a point, then it is a float.

Question: what if the json was not created in memory, but passed over the wire (from the abci app)?